tests/test_workers/test_pkg_managers/test_gomod.py

```diff
@@ -528,61 +528,118 @@ def test_go_list_cmd_failure(
+            None,
         ),
         # Directly maps to a semver non-annotated tag (v2.2.0)
-        ("/v2", "709b220511038f443fe1b26ac09c3e6c06c9f7c7", "v2.2.0"),
+        ("/v2", "709b220511038f443fe1b26ac09c3e6c06c9f7c7", "v2.2.0", None),
         # A non-semver tag (random-tag)
-        ("/v2", "37cea8ddd9e6b6b81c7cfbc3223ce243c078388a", "v2.2.1-0.20191107204245-37cea8ddd9e6"),
+        (
+            "/v2",
+            "37cea8ddd9e6b6b81c7cfbc3223ce243c078388a",
+            "v2.2.1-0.20191107204245-37cea8ddd9e6",
+            None,
+        ),
         # The go.mod file is bumped but there is no versioned commit
-        ("/v2", "6c7249e8c989852f2a0ee0900378d55d8e1d7fe0", "v2.0.0-20191108212303-6c7249e8c989"),
+        (
+            "/v2",
+            "6c7249e8c989852f2a0ee0900378d55d8e1d7fe0",
+            "v2.0.0-20191108212303-6c7249e8c989",
+            None,
+        ),
         # Three semver annotated tags on the same commit
-        ("/v2", "a77e08ced4d6ae7d9255a1a2e85bd3a388e61181", "v2.2.5"),
+        ("/v2", "a77e08ced4d6ae7d9255a1a2e85bd3a388e61181", "v2.2.5", None),
         # A non-annotated semver tag and an annotated semver tag
-        ("/v2", "bf2707576336626c8bbe4955dadf1916225a6a60", "v2.3.3"),
+        ("/v2", "bf2707576336626c8bbe4955dadf1916225a6a60", "v2.3.3", None),
         # Two non-annotated semver tags
-        ("/v2", "729d0e6d60317bae10a71fcfc81af69a0f6c07be", "v2.4.1"),
+        ("/v2", "729d0e6d60317bae10a71fcfc81af69a0f6c07be", "v2.4.1", None),
         # Two semver tags, with one having the wrong major version and the other with the correct
         # major version
-        ("/v2", "3decd63971ed53a5b7ff7b2ca1e75f3915e99cf2", "v2.5.0"),
+        ("/v2", "3decd63971ed53a5b7ff7b2ca1e75f3915e99cf2", "v2.5.0", None),
         # A semver tag that is incorrectly lower then the preceding semver tag
-        ("/v2", "0dd249ad59176fee9b5451c2f91cc859e5ddbf45", "v2.0.1"),
+        ("/v2", "0dd249ad59176fee9b5451c2f91cc859e5ddbf45", "v2.0.1", None),
         # A commit after the incorrect lower semver tag
-        ("/v2", "2883f3ddbbc811b112ff1fe51ba2ee7596ddbf24", "v2.5.1-0.20191118190931-2883f3ddbbc8"),
+        (
+            "/v2",
+            "2883f3ddbbc811b112ff1fe51ba2ee7596ddbf24",
+            "v2.5.1-0.20191118190931-2883f3ddbbc8",
+            None,
+        ),
+        # Newest semver tag is applied to a submodule, but the root module is being processed
+        (
+            "/v2",
+            "f3ee3a4a394fb44b055ed5710b8145e6e98c0d55",
+            "v2.5.1-0.20211209210936-f3ee3a4a394f",
+            None,
+        ),
+        # Submodule has a semver tag applied to it
+        ("/v2", "f3ee3a4a394fb44b055ed5710b8145e6e98c0d55", "v2.5.1", "submodule"),
+        # A commit after a submodule tag
+        (
+            "/v2",
+            "cc6c9f554c0982786ff9e077c2b37c178e46828c",
+            "v2.5.2-0.20211223131312-cc6c9f554c09",
+            "submodule",
+        ),
```

Comment:
Can we add more subpath options? Or don't we need that?

Reply:
I was thinking of adding an integration test to cover this behavior. Any ideas of what we could add to the unit test, though?

Reply:
great idea! I am not sure I got how we use submodule version tags in the test. submodule version tag is in [624 line](https://github.com/release-engineering/cachito/pull/625/files#diff-6411503dd66c67ab75b7597857d34862dd5a785a45802042b62d53df74a73c45R624)? 
And why do we have `None` instead of `submodule` in the test data sometimes? :smile_cat:

Reply:
It's easier to understand if you unpack the test repo in your computer and look at the tags there: https://github.com/release-engineering/cachito/blob/master/tests/test_workers/test_pkg_managers/golang_git_repo.tar.gz.

A normal semantic version tag looks like `v0.0.0`, which means it is applied to the root module of that repository.
A submodule tag looks like `submodule/v0.0.0`, which means it is applied to the submodule placed exaclty in the path defined in the tag (in this case, the `submodule`).

We can also have deeper nested submodules, such as `folder/anotherfolder/submodule` or even `this/is/a/very/deeply/nested/submodule`. :shrug:

Reply:
The `submodule` parameter in the tests refers to which path is currently being processed by Cachito (defined in the `packages` item, when the request is created via API).

We will only consider tags that match exactly the path being processed at the moment.